examples/typing/mg.jl

```diff
@@ -3,6 +3,149 @@ using Cassette
+Cassette.canrecurse(ctx::Type{TypeCtx},::typeof(FCollector)) = false
+function Cassette.canrecurse(ctx::TypeCtx,
+                             f::Union{typeof(+), typeof(*), typeof(/), typeof(-),typeof(Base.iterate),
+                                      typeof(Base.sum),
+                                      typeof(Base.mapreduce),
+                                      typeof(Base.Broadcast.copy),
+                                      typeof(Base.Broadcast.instantiate),
+                                      typeof(Base.Math.throw_complex_domainerror),
+                                      typeof(Base.Broadcast.broadcasted)},
+                             args...)
+    return false
+end
+                
+"""    buildgraph
+
+internal function used in the typegraphfrompath
+takes the collector object and returns a metagraph
+            
+"""
+function buildgraph(g,collector)
+    try
+        g[collector.frame.args,:label]
+    catch
+        add_vertex!(g,:label,collector.frame.args)
+    end
+
+    try 
+        g[collector.frame.ret,:label]
+    catch
+        add_vertex!(g,:label,collector.frame.ret)
+    end
+    
+    if !has_edge(g,g[collector.frame.args,:label],g[collector.frame.ret,:label])    
+        add_edge!(g,g[collector.frame.args,:label],g[collector.frame.ret,:label],:label,collector.frame.func)
+    end
+                    
+    for frame in collector.data
+        buildgraph(g,frame)
+    end
+                    
+    return g
+end
+
+"""    cleangraph(g::MetaDiGraph)
+
+removes the nothings to replace them with strings for outputing an image
+"""
+function cleangraph(mg::MetaDiGraph)
+    for vertex in vertices(mg)
+        if get_prop(mg,vertex,:label) == nothing
+            set_prop!(mg,vertex,:label,"missing")
+        end
+    end
+    
+    for edge in edges(mg)
+        if get_prop(mg,edge,:label) == nothing
+            set_prop!(mg,edge,:label,"missing")
+        end
+    end
+    
+    return mg
+end
+                            
+
+"""    typegraph(path::AbstractString,maxdepth::Int)
+            
+This is a function that takes in an array of script and produces a MetaDiGraph descibing the system.
+takes in optional parameter of recursion depth on the stacktrace defaulted to 3
+
+"""
+function typegraph(m::Module,maxdepth::Int=3)
+    
+    extractor = FCollector(maxdepth, Frame(nothing, (), nothing,)) # init the collector object     
+    ctx = TypeCtx(metadata = extractor);     # init the context we want             
+    Cassette.overdub(ctx,m.main);    # run the script internally and build the extractor data structure
+    g = MetaDiGraph()    # crete a graph where we will init our tree
+    set_indexing_prop!(g,:label)    # we want to set this metagraph to be able to index by the names
+    g = buildgraph(g,extractor)    # pass the collector ds to make the acutal metagraph
+    return cleangraph(g)
```

Comment:
you can just name this function `clean!(g::MetaGraph)` rather than `cleangraph(g::MetaGraph)`